### DISCUSSION TITLE
How to run shell commands with multiple pipes in it

### DISCUSSION THREAD

**Discussion by Ankitr19** [tokio-rs/tokio#4365] [Q&A] [answered]
I have a use case where I need to run shell commands with multiple pipes in it, for example -

`dscl . -read "/Users/#{username}" accountPolicyData |grep passwordLastSetTime -A1 |tail -1| cut -d '>' -f 2 | cut -d '<' -f 1 | cut -d . -f 1`

This is used for getting the last password change of a user, now if I pass these as args it doesn't work, what's the best way for this?

**Comment by Darksonn** [comment, marked as answer]
To do this, you have to create each process separately. To set up the pipes, you need to set the stdout of each to piped, then convert the [`ChildStdout`][2] object to a [`Stdio`][1] object that you use as the stdin of the next child. 

There's an example of how to do it in this test:

https://github.com/tokio-rs/tokio/blob/96370ba4ce9ea5564f094354579d5539af8bbc9d/tests-integration/tests/process_stdio.rs#L146-L192

[1]: https://doc.rust-lang.org/stable/std/process/struct.Stdio.html
[2]: https://docs.rs/tokio/latest/tokio/process/struct.ChildStdout.html
